### PR TITLE
Remove dependency on 'k8s.io/kubernetes' and fix version of 'k8s.io/client-go'

### PR DESCRIPTION
**Pull request by akrejcir**
The kubernetes module was not needed, and the nonexisting version of `k8s.io/client-go` was inconvenient for other projects importing this module.

```release-note
Fix dependencies, so it is more convenient to import this module.
```

### CODE REVIEW

**Comment by akrejcir** [comment]
I did not run any functional tests to check this. I'm not sure if there are any or how to run them.

**Comment by jakub-dzon** [comment]
@akrejcir - the functional tests are executed against `examples/sample-operator` using https://github.com/kubevirt/controller-lifecycle-operator-sdk/blob/master/examples/sample-operator/automation/test.sh, and they passed: https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_controller-lifecycle-operator-sdk/15/pull-controller-lifecycle-operator-sdk-e2e-k8s-1.18/1348915237394321408

File: go.mod [hunk 1]
```diff
@@ -17,17 +17,9 @@ require (
 	k8s.io/api v0.18.6
 	k8s.io/apiextensions-apiserver v0.18.6
 	k8s.io/apimachinery v0.18.6
-	k8s.io/client-go v8.0.0+incompatible
-	k8s.io/kubernetes v1.14.0
+	k8s.io/client-go v0.18.6
```
Comment (jakub-dzon) [inline go.mod:20]: Please update the dependencies in https://github.com/kubevirt/controller-lifecycle-operator-sdk/blob/master/examples/sample-operator/go.mod as well; older `k8s.io/client-go` is used there.

**Review by jakub-dzon — CHANGES_REQUESTED** [review]
Looks good; please just update the dependencies of the reference project (https://github.com/kubevirt/controller-lifecycle-operator-sdk/tree/master/examples/sample-operator)

File: go.mod (hunk 1 above)
Comment (akrejcir) [reply go.mod:20]: I'm not sure what to update there. The example project uses `github.com/operator-framework/operator-sdk v0.18.0`, and it seems it requires `k8s.io/client-go v0.18.2`.
Updating operator-sdk would need major rewrite.

Comment (jakub-dzon) [reply go.mod:20]: I meant client-go v0.18.2 to v0.18.6 - so it was aligned with the lib version.

Comment (akrejcir) [reply go.mod:20]: Ok, Done.

**Review by jakub-dzon — APPROVED** [review]
Thank you for your contribution!

**Pull request merged**